apps/src/applab/designElements/ImagePickerPropertyRow.jsx

```diff
@@ -0,0 +1,45 @@
+var React = require('react');
+
+var PropertyRow = React.createClass({
+  propTypes: {
+    initialValue: React.PropTypes.string.isRequired,
+    handleChange: React.PropTypes.func
+  },
+
+  getInitialState: function () {
+    return {
+      value: this.props.initialValue
+    };
+  },
+
+  handleChangeInternal: function (event) {
+    this.changeImage(event.target.value);
+  },
+
+  handleButtonClick: function () {
+    Applab.onDesignModeManageAssets(this.changeImage, 'image');
```

Comment:
I think that the more correct way of doing this in react is to provide an onButtonClick function as a property rather than explicitly calling into Applab from our component.